addons/account_edi_ubl_cii/models/account_move.py

```diff
@@ -121,35 +163,36 @@ def _ubl_parse_attached_document(self, tree):
         """
         attachment_node = tree.find('{*}Attachment')
         if attachment_node is None:
-            return tree
+            return '', None
 
         attachment_binary_data = attachment_node.find('./{*}EmbeddedDocumentBinaryObject')
         if attachment_binary_data is not None \
                 and attachment_binary_data.attrib.get('mimeCode') in ('application/xml', 'text/xml'):
             with suppress(etree.XMLSyntaxError, binascii.Error):
-                text = b64decode(attachment_binary_data.text)
-                return etree.fromstring(text)
+                content_1 = b64decode(attachment_binary_data.text)
+                return content_1, etree.fromstring(content_1)
 
         external_reference = attachment_node.find('./{*}ExternalReference')
         if external_reference is not None:
             description = external_reference.findtext('./{*}Description')
             mime_code = external_reference.findtext('./{*}MimeCode')
 
             if description and mime_code in ('application/xml', 'text/xml'):
+                content_2 = description.encode('utf-8')
                 with suppress(etree.XMLSyntaxError):
-                    return etree.fromstring(description.encode('utf-8'))
-        return tree
+                    return content_2, etree.fromstring(content_2)
+
+        # If neither EmbeddedDocumentBinaryObject nor ExternalReference/Description can be decoded as an XML,
+        # fall back on the contents of EmbeddedDocumentBinaryObject.
+        return content_1, None
 
     def _get_edi_decoder(self, file_data, new=False):
-        # EXTENDS 'account'
-        if file_data['type'] == 'xml':
-            if etree.QName(file_data['xml_tree']).localname == 'AttachedDocument':
-                file_data['xml_tree'] = self._ubl_parse_attached_document(file_data['xml_tree'])
-            ubl_cii_xml_builder = self._get_ubl_cii_builder_from_xml_tree(file_data['xml_tree'])
-            if ubl_cii_xml_builder is not None:
-                return ubl_cii_xml_builder._import_invoice_ubl_cii
-
-        return super()._get_edi_decoder(file_data, new=new)
+        if (ubl_model := file_data['import_file_type']) in UBL_MODELS:
```

Comment:
```py
def is_ubl(self, file_type):
    models = self.env.registry['account.edi.xml.ubl_20']._inherit_children
    while models:
        if file_type in models:
            return True
        new_models = set()
        for model in models:
            new_models.update(self.env.registry[model]._inherit_children)
        models = new_models
    return False
```
A bit technical but this avoids an arbitrary list of UBL model that has to be maintained up to date